src/task-actions.ts

```diff
@@ -291,34 +297,100 @@ export async function starknetVoyagerAction(args: TaskArguments, hre: HardhatRun
         throw new HardhatPluginError(PLUGIN_NAME, msg);
     }
     //If contract hasn't been verified yet, do it
-    let contractPath = args.path;
+    if (args.path) {
+        await handleSingleContractVerification(args.path, args.address, voyagerUrl, hre);
+    } else {
+        const splitPaths = args.paths.split(" ");
+        await handleMultiPartContractVerification(splitPaths, args.address, voyagerUrl, hre);
+    }
+}
+
+async function handleMultiPartContractVerification(
+    paths: string[],
+    address: string,
+    voyagerUrl: string,
+    hre: HardhatRuntimeEnvironment
+) {
+    paths.forEach(function (item, index) {
+        if (!path.isAbsolute(item)) {
+            paths[index] = path.normalize(path.join(hre.config.paths.root, item));
+            if (!fs.existsSync(paths[index])) {
+                throw new HardhatPluginError(PLUGIN_NAME, `File ${paths[index]} does not exist`);
+            }
+        }
+    });
+
+    const bodyFormData = new URLSearchParams();
```

Comment:
I find it unintuitive that for `bodyFormData` is constructed using `URLSearchParams`. Is this really the way?

Reply:
the other option for the formData would be to add a new dependency 'form-data', but the URLSearchParams works exactly the same